app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -40,7 +40,7 @@ def confirmation_path(next_mfa_selection_choice = nil)
 
   def confirm_user_authenticated_for_2fa_setup
     authenticate_user!(force: true)
-    return if user_fully_authenticated?
+    return if user_fully_authenticated? && !UserSessionContext.reauthentication_context?(context)
```

Comment:
does this need to be conditional based on the feature flags?

Reply:
yep, fixed in cc9e638f7915bf97e0fbe7b28d7197d8f016f629